Update-AutomationAzureModulesForAccount.ps1

```diff
@@ -253,26 +291,49 @@ function AreAllModulesAdded([string[]] $ModuleListToAdd) {
 # element in the array consist of modules with no dependencies.
 # The second element only depends on the modules in the first element, the
 # third element only dependes on modules in the first and second and so on. 
-function Create-ModuleImportMapOrder {
+function Create-ModuleImportMapOrder([bool] $AzModuleOnly) {
     $ModuleImportMapOrder = $null
-    # Get the latest version of the AzureRM.Profile module
-    $VersionAndDependencies = Get-ModuleDependencyAndLatestVersion $script:AzureRMProfileModuleName
+    $ProfileOrAccountsModuleName = $null
+    $GetAutomationModule = $null
 
-    $AzureRMProfileEntry = $script:AzureRMProfileModuleName
-    $AzureRMProfileEntryArray = ,$AzureRMProfileEntry
-    $ModuleImportMapOrder += ,$AzureRMProfileEntryArray
+    # Use the relevant module class to avoid conflicts
+    if ($AzModuleOnly) {
+        $ProfileOrAccountsModuleName = $script:AzAccountsModuleName
+        $GetAutomationModule = $script:GetAzAutomationModule
+    } else {
+        $ProfileOrAccountsModuleName = $script:AzureRmProfileModuleName
+        $GetAutomationModule = $script:GetAzureRmAutomationModule
+    }
 
     # Get all the modules in the current automation account
-    $CurrentAutomationModuleList = Get-AzureRmAutomationModule `
+    $CurrentAutomationModuleList = & $GetAutomationModule `
                                         -ResourceGroupName $ResourceGroupName `
                                         -AutomationAccountName $AutomationAccountName
 
+    # Get the latest version of the AzureRM.Profile OR Az.Accounts module
+    $VersionAndDependencies = Get-ModuleDependencyAndLatestVersion $ProfileOrAccountsModuleName
+
+    $ModuleEntry = $ProfileOrAccountsModuleName
+    $ModuleEntryArray = ,$ModuleEntry
+    $ModuleImportMapOrder += ,$ModuleEntryArray
+
     do {
         $NextAutomationModuleList = $null
         $CurrentChainVersion = $null
         # Add it to the list if the modules are not available in the same list 
         foreach ($Module in $CurrentAutomationModuleList) {
             $Name = $Module.Name
+
+            # ignore conflicting modules
+            if ($AzModuleOnly) {
+                if ($Name -eq "AzureRM" -Or $Name -Like "AzureRM.*") {
+                    continue

```

Comment:
When I invoke this runbook with 'Az', it now gets stuck in `Create-ModuleImportMapOrder` looping infinitely, most likely because of the `while ($null -ne $CurrentAutomationModuleList)` condition. Perhaps the incompatible modules should be filtered out earlier, e.g. on line 309:
``` PowerShell
$CurrentAutomationModuleList = & $GetAutomationModule ... |
    ?{
        ($AzModuleOnly -and ($_.Name -eq 'Az' -or $_.Name -like 'Az.*)) -or
        (!$AzModuleOnly -and ($_.Name -eq 'AzureRM' -or $_.Name -like 'AzurerRM.*)))
    }
```

Reply:
Very interesting, I didn't know about this type of notation.

What about modules like [Azure.Storage](https://www.powershellgallery.com/packages/Azure.Storage/4.6.1)? It looks like this will be unnecessarily filtered out by the above.

My guess is you would instead need the additional dot wildcard match for Azure too.
`($_.Name -eq 'Azure' -or $_.Name -like 'Azure.*')`

Reply:
Yes, you are right, I forgot about Azure.*, your fix looks correct.

Reply:
Corrected and pushed, but untested as I don't have access to Azure until next Thursday.

Reply:
@AnatoliB This still does not work.

So after some testing and analysis, I noticed that Azure.Storage has AzureRM dependencies. This probably means that Azure modules are also not compatible with Az. 

This works, but is not so robust.
(!$AzModuleOnly -and ($_.Name -eq 'AzureRM' -or $_.Name -like 'AzureRM.*' -or $_.Name -eq 'Azure' -or $_.Name -like 'Azure.*')).

Reply:
@ayoung012 Thank you, I forgot that **Azure.Storage** actually depends on **AzureRM**, and **Azure** also depends on **Azure.Storage**. In this case, they should be treated the same way as **AzureRM** modules, and your fix looks correct to me. The alternative would be to write code analyzing these dependencies dynamically. However, given the fact that the development of **AzureRM** is stopped, this looks like an overkill, so having this logic hardcoded here is not that bad.

So, if this works now, please go ahead and submit the change.

Reply:
Would be nice to analyse dependencies dynamically, but I think in this case it would be overkill.

Changes have been submitted.